### DISCUSSION TITLE
Using qnap container station to run auto-southwest-check-in in docker container

### DISCUSSION THREAD

**Discussion by Arcanlaw** [jdholtz/auto-southwest-check-in#125] [General]
I have zero experience with containers, so bear with me:

I successfully created a container following these steps:

QNapp AppCenter -> Container Station -> (+) Create -> Search for jdholtz/auto-southwest-check-in -> Install

A "Create Container" popup then shows up:

![image](https://github.com/jdholtz/auto-southwest-check-in/assets/77711984/3bb245eb-c7a4-42b8-b098-dd466702743e)

I click on Create and it is created in a few seconds:

![image](https://github.com/jdholtz/auto-southwest-check-in/assets/77711984/6e8fa160-fc31-4cff-b171-aaed09a9ac41)

But here, I am stuck.  If I click "Start" it appears to run for about 2 seconds and then shows Stopped again.  I can't type anything in the Console.

I'm sure I'm doing something elementary and silly, would appreciate if someone can point it out for me!

**Comment by jdholtz** [comment]
I’ve never used Qnap, but I think the reason why it exists after two seconds is that it doesn’t have any reservations or accounts to monitor. Are you either attaching a configuration file in the container or passing the information through the container’s entrypoint?

**Reply by Arcanlaw** [reply]
Thanks for the quick response.  I'm not passing anything, just creating the container and figured there would be some way to enter that command after it was created.  In the first screenshot, there's a field for "command" ... might that be where I enter the reservation?  I have a few flights lined up over the next several weeks, would I then be spinning up a different container for every reservation?  Any guesses on the format of the 'command' field, and maybe more importantly, is there some way of testing the script/container so I can have a reasonable assumption that it might work?  Thank you!

**Reply by jdholtz** [reply]
If you have multiple reservations, you can run them all through the same container using the configuration file. It appears you can attach it through the `Volumes` tab that I see in your screenshot. You can check out the Configuration.md doc in this repository for information on how to set up the config for multiple reservations/accounts.

There isn't a way to test the script, but just running it should let you know if it will work correctly because you should see it schedule all of your flights.

**Reply by Arcanlaw** [reply]
Ok I think I have a handle on this now - thank you!

**Reply by jwin3** [reply]
I got it to work after spending a couple of hours on it. The biggest issue I had was running the code so it would load up my configuration file. In QNAP's container station, it was having issues, so you have to use an ssh protocol through an application called putty. After you download putty, type in your ip address for your NAS, then login with your username and password, you'll get a console management screen, quit to return to normal shell enviroment. Now in the putty shell, type in "docker run -d --restart on-failure --volume /share/Container/SouthwestCheckIN/config.json:/app/config.json jdholtz/auto-southwest-check-in ARGS" Your config file location will be in a different location, so modify that slightly.
